### PR TITLE
refactor(Telemetry): Ensure to report `projectId` for interactive

### PR DESCRIPTION
**Pull request by pgrzesik**
Reported internally:
- Ensure to report `projectId` for interactive flow deployments
- Ensure to report creation of a new stack during deployment

### CODE REVIEW

File: scripts/serverless.js [hunk 1]
```diff
@@ -558,6 +559,7 @@ const processSpanPromise = (async () => {
               configuration,
               commandUsage,
               variableSources: variableSourcesInConfig,
+              interactiveContext,
```
Comment (medikoo) [inline scripts/serverless.js:562]: Wouldn't it be cleaner to expose `serverless` instance on interactive CLI context, and pass it to `.generateTelemetryPayload` (having that we probably don't have to make any new changes in `generateTelemetryPayload`)?

Comment (pgrzesik) [reply scripts/serverless.js:562]: That's definitely an option and might be a better solution overall 👍

File: lib/utils/telemetry/generatePayload.js [hunk 2]
```diff
@@ -300,17 +300,19 @@ module.exports = ({
...
-      const accountId = serverless.getProvider('aws').accountId;
+      const accountId = serverless && serverless.getProvider('aws').accountId;
       if (serviceName && accountId) {
         payload.projectId = crypto
           .createHash('sha256')
           .update(`${serviceName}-${accountId}`)
           .digest('base64');
       }
+
+      payload.didCreateStack = Boolean(serverless && serverless.getProvider('aws').didCreateStack);
```
Comment (mnapoli) [inline lib/utils/telemetry/generatePayload.js:315]: Slightly related: we mentioned the stack deletion as well, and we said "we can look at `remove` commands" -> but these don't have the `projectId` property in those events. Should we add `command === 'remove'` above in the `if` too? (or maybe in a separate PR if that's cleaner)

Comment (pgrzesik) [reply lib/utils/telemetry/generatePayload.js:315]: That's a great point - I will add that in a separate PR

File: scripts/serverless.js [hunk 3]
```diff
@@ -514,6 +514,7 @@ const processSpanPromise = (async () => {
     const isStandaloneCommand = notIntegratedCommands.has(command);
 
     if (!isHelpRequest && (isInteractiveSetup || isStandaloneCommand)) {
+      let interactiveContext;
```
Comment (medikoo) [inline scripts/serverless.js:517]: It'll be nicer to reuse the top level `serverless` variable (same as we reuse `configuration` one).

File: lib/utils/telemetry/generatePayload.js (hunk 2 above)
Comment (medikoo) [inline lib/utils/telemetry/generatePayload.js:315]: In platform PR I've suggested that maybe it'll be better to give it a more agnostic name as `didCreateService`, so if eventually later we will support non-CF-related deployments, this name stays accurate.

WDYT?

Comment (mnapoli) [reply lib/utils/telemetry/generatePayload.js:315]: Not sure if that applies to the metric name as well, but for the metric name +1 to have it a bit more generic

Comment (pgrzesik) [reply lib/utils/telemetry/generatePayload.js]: Great suggestion @medikoo 👍

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**